model_api/python/model_api/models/action_classification.py

```diff
@@ -0,0 +1,220 @@
+"""
+ Copyright (C) 2024 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+from model_api.adapters.utils import RESIZE_TYPES, InputTransform
+
+from .model import Model
+from .types import BooleanValue, ListValue, NumericalValue, StringValue
+from .utils import ClassificationResult, load_labels
+
+if TYPE_CHECKING:
+    from model_api.adapters.inference_adapter import InferenceAdapter
+
+
+class ActionClassificationModel(Model):
+    """An wrapper for an action classification model
+
+    The ActionClassificationModel has 1 or more inputs with images - 6D tensors.
```

Comment:
Is that true from wrapper user's perspective?

Reply:
I found that this class assumes input should be single video clip w/ 4D tensors. I updated it https://github.com/openvinotoolkit/model_api/pull/178/commits/7e4dd83d40c20d88ab7dbe19f42315cb2e21dc2d